esphome/components/climate/__init__.py

```diff
@@ -119,10 +119,19 @@
 )
 
 
-def single_visual_temperature(value):
+def visual_temperature_step(value):
+    VISUAL_TEMPERATURE_STEP_SCHEMA = cv.Schema(
+        {
+            cv.Required(CONF_TARGET_TEMPERATURE): visual_temperature,
+            cv.Required(CONF_CURRENT_TEMPERATURE): visual_temperature,
+        }
+    )
```

Comment:
The schema should be outside the function as we dont want to recreate it multiple times
```suggestion
VISUAL_TEMPERATURE_STEP_SCHEMA = cv.Schema(
    {
        cv.Required(CONF_TARGET_TEMPERATURE): visual_temperature,
        cv.Required(CONF_CURRENT_TEMPERATURE): visual_temperature,
    }
)

def visual_temperature_step(value):
```

Reply:
Thanks, Jesse - makes sense!